### PR TITLE
[Core Lich] Ruby 3 compatibility

### PR DESCRIPTION
**Pull request by MahtraDR**
This has been well tested on my end, but I haven't kept it up to date with any core changes since I created the original branch.

Also, for posterity, ... 
image

### CODE REVIEW

File: lich.rbw [hunk 1]
```diff
@@ -100,12 +100,6 @@ class Numeric
 end
 
 class String
-  @@elevated_untaint = proc { |what| what.orig_untaint }
```
Comment (rpherbig) [inline lich.rbw:103]: 💡 There's still a lot of 'elevated' variable naming floating around. Not functional, but probably could be renamed in the future to make more sense?

File: lich.rbw [hunk 2]
```diff
@@ -2306,7 +2205,7 @@ module Games
                   $_SERVERSTRING_.gsub!("\r\n", "</component>")
                   Lich.log "Open-ended room objects component id tag fixed to: #{$_SERVERSTRING_.inspect}"
                 end
-                # "</component>\r\n"                
+                # "</component>\r\n"
```
Comment (rpherbig) [inline lich.rbw:2208]: 💡 Delete?

**Review by rpherbig — APPROVED** [review]
Seems reasonable to me!

File: lich.rbw [hunk 3]
```diff
@@ -3236,7 +3132,6 @@ module Games
     end
 
     require_relative("./lib/spell.rb")
-    require_relative("./lib/bounty.rb")
```
Comment (mrhoribu) [inline lich.rbw:3239]: Looks like this was accidentally removed on PR to merge into existing code base. Need to add back in.

Comment (MahtraDR) [reply lich.rbw]: Re-added. Thank you!

File: lich.rbw (hunk 2 above)
Comment (MahtraDR) [reply lich.rbw:2208]: Removed. Thank you!

File: lich.rbw (hunk 1 above)
Comment (MahtraDR) [reply lich.rbw:103]: I agree. That just makes it a bigger change for now.

**Comment by MahtraDR** [comment]
> Seems reasonable to me!

Thank you for taking the time to review this. I appreciate it.

**Comment by MahtraDR** [comment]
This version brings us up to compatibility with Ruby 3.2.

**Pull request merged**